### PR TITLE
Add JUnitXmlReporter

### PR DESCRIPTION
**Pull request by nowsprinting**
### Changes

- Add `JUnitXmlReporter` (based on  internal `JUnitReporter` class)
- Remove `junitReportPath` from `AutopilotSettings`, and add convert to `JUnitXmlReporter`
- Mod test-helper package version for running tests

### Priority

I hope to your review && merge around one week.
There is no need to release it yet.

---

### Contribution License Agreement

- [x] By placing an "x" in the box, I hereby understand, accept and agree to be bound by the terms and conditions of the [Contribution License Agreement](https://dena.github.io/cla/).

### CODE REVIEW

**Comment by get-me-power** [comment]
@nowsprinting 

`Runtime/Settings/AutopilotSettings.cs` is conflicted.
Will you resolve conflicts?

**Comment by nowsprinting** [comment]
@get-me-power san, rebased!

**Review by asurato — APPROVED** [review]
Although I commented a little, LGTM.

File: Editor/UI/Reporters/JUnitXmlReporterEditor.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        private const float SpacerPixels = 10f;
+
+        // @formatter:off
+        private static readonly string s_description = L10n.Tr("Description");
+        private static readonly string s_descriptionTooltip = L10n.Tr("Description about this Reporter instance");
+        private GUIContent _descriptionLabel;
+        private SerializedProperty _descriptionProp;
+
+        private static readonly string s_outputPath = L10n.Tr("Output File Path");
+        private static readonly string s_outputPathTooltip = L10n.Tr("Relative path from the project root directory. When run on player, it will be the Application.persistentDataPath.");
```
[the remark this answers is not in the document]
Comment (nowsprinting) [reply Editor/UI/Reporters/JUnitXmlReporterEditor.cs:25]: Maybe you have misunderstood about "player".

"Player" is a general term for iOS/Android devices, Windows PCs, etc. It does not mean "human" or "manual."

Regarding "it", should I do the following?

```suggestion
        private static readonly string s_outputPathTooltip = L10n.Tr("Relative path from the project root directory. When run on player, relative path from the Application.persistentDataPath.");
```

[the remark this answers is not in the document]
Comment (nowsprinting) [reply Editor/UI/Reporters/JUnitXmlReporterEditor.cs:25]: Note: This message will be changed in the next planned update, "Add output root path field into AutopilotSettings."
see #100

[the remark this answers is not in the document]
Comment (asurato) [reply Editor/UI/Reporters/JUnitXmlReporterEditor.cs:25]: > "Player" is a general term for iOS/Android devices, Windows PCs, etc. It does not mean "human" or "manual."
Sorry, I mistook.

> Regarding "it", should I do the following?
It's OK.

> Note: This message will be changed in the next planned update, "Add output root path field into AutopilotSettings."
I see. I will merge it soon after.

**Comment by asurato** [comment]
@nowsprinting 
Thank you for your contribution!

**Comment by asurato** [comment]
@nowsprinting 
> `Relative path from the project root directory. When run on player, relative path from the Application.persistentDataPath.`

The latter sentence make sense slightly, but the structure of the sentence is broken.
Please consider to use `Relative path from the project root directory or the Application.persistentDataPath in runs on player.` if you see this comment.

**Comment by nowsprinting** [comment]
Thanks, @asurato san,

Change the relative-path-root to "output root path field in AutopilotSetings", later.
This makes it easier to write.
However, I want to express that absolute paths can also be specified.

**Pull request merged**